components/sidebar/sidebar_channel/sidebar_channel_icon/sidebar_channel_icon.tsx

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import ArchiveIcon from 'components/widgets/icons/archive_icon';
+import DraftIcon from 'components/widgets/icons/draft_icon';
+
+type Props = {
+    icon?: JSX.Element;
+    channelIsArchived: boolean;
+    hasDraft: boolean;
+};
+
+export default class SidebarChannelIcon extends React.PureComponent<Props> {
+    render() {
```

Comment:
Why do we have two separate pieces of logic for determining the icon of a channel? Perhaps we could combine this with the logic from the SidebarBaseChannel?

Reply:
Looking at the SidebarDirectChannel now, and we could also combine that logic with this as well. Perhaps we could structure it something like having `SidebarChannelIcon` look at the `channel` and conditionally render components like `GlobeIcon`, `ArchiveIcon`, or a more complicated `DirectMessageIcon` that could do things like getting the teammate's status and render a bot's icon instead of having that logic distributed between so many different places.

Reply:
As discussed, we're leaving it up to the `SidebarChannel` components to pass in their icons